chore(readme): enhance warning

realguse opened the pull request:
This PR enhances the README.md by replacing the old warning with a warning that looks nicer.

**Before**:
![image](https://github.com/user-attachments/assets/ab17bb65-1dbd-4679-86cf-40cd802ab92c)

**After**:
![image](https://github.com/user-attachments/assets/a135d3dc-223b-432a-ae87-ec80893f79f8)

janhohenheim commented:
Don’t know if this readme is included in any of our docs, but if it is, this will trip up cargo doc, as it will parse it as a hyperlink FYI

realguse commented:
This is just the deafult README.md with the "WARNING" message being replaced.

janhohenheim commented:
I know :) You can do `#![doc = include_str!("../README.md")]` to include the readme in your rust docs, which lots of crates do. I don't think we do this anywhere in Bevy, but if we did, rust doc would not like the `![WARNING]` syntax, as it would resolve to an incomplete link.

realguse commented:
Could you give me the link to the docs where the README is present

realguse commented:
I tested this on other markdown renderers, and there doesn't seem to be any other place than github that renders it.

janhohenheim commented:
Sure! LWIM uses this technique:
- [README](https://github.com/Leafwing-Studios/leafwing-input-manager/blob/main/README.md)
- [doc include attribute](https://github.com/Leafwing-Studios/leafwing-input-manager/blob/main/src/lib.rs#L3)
- [docs.rs page showing the README](https://docs.rs/leafwing-input-manager/latest/leafwing_input_manager/)

Yeah, GitHub is the only place that accepts the `![WARNING]` etc. syntax, the other renderers will treat it as a simple (unresolved) link.

realguse commented:
There seems to be some confusion here. This PR is for the README.md in [this repo](https://github.com/bevyengine/bevy/blob/main/README.md).

janhohenheim commented:
As I said, I don't think Bevy includes the readme anywhere. LWIM is just an example of how it would look like if it did. 
I'm just saying that before merging this, we should make sure that the readme is never included anywhere as documentation.

realguse commented:
Oh, i'm so sorry, ill see if i find anything.

SludgePhD commented:
crates.io render the readme too: https://crates.io/crates/bevy

MrGVSV commented on README.md:
```diff
@@ -11,12 +11,8 @@
-**MSRV:** Bevy relies heavily on improvements in the Rust language and compiler.
-As a result, the Minimum Supported Rust Version (MSRV) is generally close to "the latest stable release" of Rust.
+> [!WARNING]
+> Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing breaking changes to the API is released [approximately once every 3 > months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). We provide [migration guides](https://bevyengine.org/learn/migration-guides/), but we can't guarantee migrations ...
```
Typo: `>`
```suggestion
> Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing breaking changes to the API is released [approximately once every 3 > months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). We provide [migration guides](https://bevyengine.org/learn/migration-guides/), but we can't guarantee migrations will always be easy. Use only if you are willing to work in this environment.
```

MrGVSV commented on README.md:
```diff
@@ -11,12 +11,8 @@
-
-Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing breaking changes to the API is released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). We provide [migration guides](https://bevyengine.org/learn/migration-guides/), but we can't guarantee migrations wil ...
-
-**MSRV:** Bevy relies heavily on improvements in the Rust language and compiler.
-As a result, the Minimum Supported Rust Version (MSRV) is generally close to "the latest stable release" of Rust.
```
This probably shouldn't have been removed. Could you add it back?

realguse replied:
Sure, but I don't think theres an option for a newline in the warning alert, so it wiil have to be on the same line @MrGVSV.

MrGVSV reviewed (commented):
I do like using these new annotations. They're prettier and easier on the eyes.

Assuming this README isn't used as documentation on docs.rs or any other renderer, I think we should be fine to use this.

Just a couple typos/suggestions.

realguse commented:
I've gone ahead and fixed the type (`>`) and added the rust version warning back.

I tried to short it down, but let me know if i should use the original.

realguse commented:
> crates.io render the readme too: https://crates.io/crates/bevy

@SludgePhD, do you know if crates.io can render alerts?

MrGVSV commented on README.md:
```diff
@@ -11,12 +11,8 @@
-
-Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing breaking changes to the API is released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). We provide [migration guides](https://bevyengine.org/learn/migration-guides/), but we can't guarantee migrations wil ...
-
-**MSRV:** Bevy relies heavily on improvements in the Rust language and compiler.
-As a result, the Minimum Supported Rust Version (MSRV) is generally close to "the latest stable release" of Rust.
```
I think we should keep the text the same since it also indicates why we tend to stick to latest stable.

For the newlines I think you can do:
```md
> some text.
>
> some more text
```

realguse replied:
Thanks, I'll do it now!

realguse replied:
![image](https://github.com/user-attachments/assets/a135d3dc-223b-432a-ae87-ec80893f79f8)

Dont it!

SludgePhD commented:
> @SludgePhD, do you know if crates.io can render alerts?

I don't know, but they are a non-standard feature that isn't supported by most standard markdown engines, so if I had to guess I'd say no.

realguse commented:
> > @SludgePhD, do you know if crates.io can render alerts?
> 
> I don't know, but they are a non-standard feature that isn't supported by most standard markdown engines, so if I had to guess I'd say no.

Is there any way of excluding it from the crates.io readme?

SludgePhD commented:
I don't think so. The only thing you can do is to have two readmes, maybe one of them generated from the other to reduce duplication. Too much hassle for this minor change for sure. Or you could remove the crates.io readme entirely.

janhohenheim commented:
Or set up the `[WARNING]` to be a valid link, but idk what it would link to. Rick Roll? :p 
Overall, I'm against merging this until we can make sure it does not break our crates.io presence.

realguse commented:
The warning could link to the github readme, or (if its possibly) link to the warning alert in the github readme

realguse commented:
It could link to a URL like this:
https://github.com/realguse/bevy#:~:text=Bevy%20is%20still%20in%20the%20early%20stages%20of%20development.%20Important%20features%20are%20missing.%20Documentation%20is%20sparse.%20A%20new%20version%20of%20Bevy%20containing%20breaking%20changes%20to%20the%20API%20is%20released%20approximately%20once%20every%203%20months.%20We%20provide%20migration%20guides%2C%20but%20we%20can%27t%20guarantee%20migrations%20will%20always%20be%20easy.%20Use%20only%20if%20you%20are%20willing%20to%20work%20in%20this%20environment.

Which will highlight the warning.

realguse commented:
I tried linking to the higlitied text but it turned out like this:

![image](https://github.com/user-attachments/assets/21b5abec-3e1f-496a-966f-9ad118c9be17)

realguse commented:
Closed because of crates.io.
